snykTask/src/task-args.ts

```diff
@@ -102,31 +111,45 @@ class TaskArgs {
     return this.projectName;
   }
 
+  // validate based on testTypeSeverityThreshold applicable thresholds
   public validate() {
-    if (this.failOnThreshold) {
-      if (this.isNotValidThreshold(this.failOnThreshold)) {
-        const errorMsg = `If set, failOnThreshold must be '${Severity.CRITICAL}' or '${Severity.HIGH}' or '${Severity.MEDIUM}' or '${Severity.LOW}' (case insensitive). If not set, the default is '${Severity.LOW}'.`;
-        throw new Error(errorMsg);
-      }
+    const taskTestType = this.testType || TestType.APPLICATION;
+    const validTestTypes: any = Object.values(TestType);
+    const taskTestTypeThreshold =
+      testTypeSeverityThreshold.get(taskTestType) ??
+      testTypeSeverityThreshold.get(TestType.APPLICATION);
+
+    if (!validTestTypes.includes(taskTestType)) {
+      const warningMsgTestType = `Invalid testType specified. Defaulted to 'app'`;
+      console.log(warningMsgTestType);
     }
 
-    if (this.severityThreshold) {
-      if (this.isNotValidThreshold(this.severityThreshold)) {
-        const errorMsg = `If set, severityThreshold must be '${Severity.CRITICAL}' or '${Severity.HIGH}' or '${Severity.MEDIUM}' or '${Severity.LOW}' (case insensitive). If not set, the default is '${Severity.LOW}'.`;
+    if (this.failOnThreshold) {
+      if (
+        !taskTestTypeThreshold?.includes(this.failOnThreshold.toLowerCase())
+      ) {
+        const errorMsg = `If set, failOnThreshold must be one from [${taskTestTypeThreshold}] (case insensitive). If not set, the default is '${Severity.LOW}'.`;
         throw new Error(errorMsg);
       }
     }
-  }
 
-  private isNotValidThreshold(threshold: string) {
-    const severityThresholdLowerCase = threshold.toLowerCase();
+    if (
+      taskTestType != TestType.CODE &&
+      this.severityThreshold &&
+      !taskTestTypeThreshold?.includes(this.severityThreshold.toLowerCase())
+    ) {
+      const errorMsg = `If set, severityThreshold must be one from [${taskTestTypeThreshold}] (case insensitive). If not set, the default is '${Severity.LOW}'.`;
+      throw new Error(errorMsg);
+    }
 
-    return (
-      severityThresholdLowerCase !== Severity.CRITICAL &&
-      severityThresholdLowerCase !== Severity.HIGH &&
-      severityThresholdLowerCase !== Severity.MEDIUM &&
-      severityThresholdLowerCase !== Severity.LOW
-    );
+    if (
+      taskTestType == TestType.CODE &&
+      this.codeSeverityThreshold &&
+      !taskTestTypeThreshold?.includes(this.codeSeverityThreshold.toLowerCase())
+    ) {
+      const errorMsg = `If set, codeSeverityThreshold must be one from [${taskTestTypeThreshold}] (case insensitive). If not set, the default is '${Severity.LOW}'.`;
+      throw new Error(errorMsg);
+    }
```

Comment:
I noticed that we are using loose equality checks (`==` and `!=`) over strict equality. Would it be okay if we replace them with `!==` and `===`? [Strict equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness) checks ensure that both the **type** and **value** must match, leading to more predictable and safer code.

Reply:
corrected with strict equality